packages/edge-gateway/src/gateway.js

```diff
@@ -43,3 +57,25 @@ function getTransformedResponseWithCustomHeaders(response) {
 
   return clonedResponse
 }
+
+/**
+ * Get a given entry from the goodbits list if CID exists, and return tags
+ *
+ * @param {KVNamespace} datastore
+ * @param {string} cid
+ */
+async function getTagsFromGoodbitsList(datastore, cid) {
+  if (!datastore || !cid) {
+    return []
+  }
+
+  // TODO: Remove once https://github.com/nftstorage/nftstorage.link/issues/51 is fixed
+  const goodbitsEntry = await pRetry(() => datastore.get(cid), { retries: 5 })
```

Comment:
I think we are always going to need retries at this scale. Even if they fix some root issue, we will still see the occasional random failure that can be fixed by asking again. Pesky entropy.